### PR TITLE
feat(zscript): merged npc->flags and npc->flags2 into npc->flags[]

### PR DESCRIPTION
**Pull request by ghost**
gotta go through the zasm stuff still but here's progress

### CODE REVIEW

**Comment by connorjclark** [comment]
Looking good!

A good way to split this would be all this refactor stuff upfront (which is nearly done here), followed by just the zscript additions. Smaller PRs are easier to review, gets progress out faster, and allows us to be more specific with the more user-facing change in the changelog (a `refactor` commit is less likely to be read to learn about "new zscript features!" than a `feat(zscript)` commit)

**Comment by ghost** [comment]
Squashed

File: src/zc/ffscript.cpp [hunk 1]
```diff
@@ -13522,6 +13520,17 @@ int32_t get_register(int32_t arg)
 			break;
 		}
```
Comment (connorjclark) [inline src/zc/ffscript.cpp:13522]: missing NPCDATAFLAGS2, and NPCDATAFLAGS is wrong (its being used as an array here). NPCDATAFLAGS needs to return the lower half of `flags` (but *10000), and NPCDATAFLAGS2 should be the upper half.

File: src/zc/ffscript.cpp [hunk 2]
```diff
@@ -47667,15 +47715,17 @@ void FFScript::write_enemies(PACKFILE *f, int32_t vers_id)
...
+		if(!p_iputll(guysbuf[i].flags,f))
 		{
 		Z_scripterrlog("do_savegamestructs FAILED to write GUY NODE: %d",6);
 		}
 		
+		/*
 		if(!p_iputl(guysbuf[i].flags2,f))
 		{
 		Z_scripterrlog("do_savegamestructs FAILED to write GUY NODE: %d",7);
 		}
```
Comment (connorjclark) [inline src/zc/ffscript.cpp]: can just delete

**Review by connorjclark — DISMISSED** [review]
100 more PRs like this and we might have a sensible code base.

File: src/zc/ffscript.cpp (hunk 1 above)
Comment (ghost) [reply src/zc/ffscript.cpp:13522]: give me a hour or two

Comment (connorjclark) [reply src/zc/ffscript.cpp:13522]: (talked on discord)

forgetaabouttt it. we killing these 🔥 (cuz _no_ quests use them)

**Comment by ghost** [comment]
You got it
________________________________

**Comment by Jambu1989** [comment]
your still requesting changes i took care of the comments. what else?

File: src/zc/ffscript.cpp [hunk 3]
```diff
@@ -39465,9 +39483,8 @@ int32_t run_script_int(bool is_jitted)
 			case GETNPCDATATILE: FFScript::getNPCData_tile(); break;
 			case GETNPCDATAEHEIGHT: FFScript::getNPCData_e_height(); break;
 			case GETNPCDATAFLAGS: FFScript::getNPCData_flags(); break;
-			case GETNPCDATAFLAGS2: FFScript::getNPCData_flags2(); break;
-			case GETNPCDATAWIDTH: FFScript::getNPCData_flags2(); break;
-			case GETNPCDATAHEIGHT: FFScript::getNPCData_flags2(); break;
```
Comment (connorjclark) [inline src/zc/ffscript.cpp:39470]: lol

File: src/zc/ffscript.cpp [hunk 4]
```diff
@@ -42608,6 +42630,17 @@ void FFScript::getNPCData_misc()
 	else set_register(sarg1, guysbuf[ID].attributes[indx] * 10000);
 }
 
+void FFScript::getNPCData_flags()
+{
+	int32_t ID = int32_t(ri->d[rINDEX] / 10000); //the enemy ID value
+	int32_t indx = int32_t(ri->d[rINDEX2] / 10000); //the misc index ID
+	if ((ID < 1 || ID > 511) || (indx < 0 || indx > 63))
+		set_register(sarg1, -10000);
+	else set_register(sarg1, (guysbuf[ID].flags<<indx )* 10000);
```
Comment (connorjclark) [inline src/zc/ffscript.cpp:42639]: this should be `(flags & LLF(indx)) ? 10000 : 0` right?

**Pull request closed without merging**